sdk/storage/azure-storage-queue/azure/storage/queue/_models.py

```diff
@@ -286,6 +295,8 @@ def _extract_data_cb(self, messages): # pylint: disable=no-self-use
         # There is no concept of continuation token, so raising on my own condition
         if not messages:
             raise StopIteration("End of paging")
+        if self._max_messages is not None:
+            self._max_messages = self._max_messages - self.results_per_page
```

Comment:
self._max_messages = self._max_messages - len(messages)?

Reply:
Good catch! I had this in my notes but completely missed it when moving this logic :) thanks Anna!